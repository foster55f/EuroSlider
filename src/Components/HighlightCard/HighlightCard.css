.game-card{
    background-color: black;
}