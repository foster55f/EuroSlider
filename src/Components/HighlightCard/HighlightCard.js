import React from 'react';


const HighlightCard = ({ title, embed, id }) => {
    return (
        <article className='game-card'>
            <h1>{title.toUpperCase()}</h1>
            <div dangerouslySetInnerHTML={{__html: `${embed}`}} />  
            {/* <div className='game-img' style={{ backgroundImage: `url(${image})` }} /> */}
      </article>
    )
}

export default HighlightCard