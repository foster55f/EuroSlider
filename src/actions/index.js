export const addGames = games => ({
    type: 'ADD_GAMES',
    games
})

export const filterGames = displayGames => ({
    type: 'FILTER_GAMES',
    displayGames
})

