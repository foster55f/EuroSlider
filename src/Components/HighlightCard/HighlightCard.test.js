import React from 'react';
import { shallow } from 'enzyme';
import { HighlightCard, mapStateToProps, mapDispatchToProps } from './HighlightCard';
import { addFavoriteGames, addUserFavoriteGames, removeGames, removeUserGames } from '../../actions';


describe('HighlightCard', () => {
  let wrapper;

  it('should match the App snapshot', () => {
    wrapper = shallow(<HighlightCard />);
    expect(wrapper).toMatchSnapshot();
  })


  describe('mapsStateToProps', () => {
    it('should return user info and games from the store', () => {
        const mockState = {
            user: {},
            games: [],
            favoriteGames: [],
            userLog: false,
        };
        const expected = {
            user: {},
            games: [],
            favoriteGames: [],
            userLog: false,
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
    });
});

describe('mapDispatchToProps', () => {
    it('calls dispatch with an addFavoriteGames action when addFavoriteGames is called',
        () => {
            const mockDispatch = jest.fn();
            const actionToDispatch = addFavoriteGames(['barcelona']);
            const mappedProps = mapDispatchToProps(mockDispatch);

            mappedProps.addFavoriteGames(['barcelona']);

            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      });
  
    it('calls dispatch with an addFavoriteGames action when addFavoriteGames is called',
      () => {
          const mockDispatch = jest.fn();
          const actionToDispatch = addUserFavoriteGames(['barcelona']);
          const mappedProps = mapDispatchToProps(mockDispatch);

          mappedProps.addUserFavoriteGames(['barcelona']);

          expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
          });
    
    it('calls dispatch with an removeGames action when removeGames is called',
          () => {
              const mockDispatch = jest.fn();
              const actionToDispatch = removeGames(['barcelona']);
              const mappedProps = mapDispatchToProps(mockDispatch);
    
              mappedProps.removeGames(['barcelona']);
    
              expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
        });
    it('calls dispatch with an removeUserGames action when removeUserGames is called',
        () => {
            const mockDispatch = jest.fn();
            const actionToDispatch = removeUserGames(['barcelona']);
            const mappedProps = mapDispatchToProps(mockDispatch);
  
            mappedProps.removeUserGames(['barcelona']);
  
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      });
 
  });  
})