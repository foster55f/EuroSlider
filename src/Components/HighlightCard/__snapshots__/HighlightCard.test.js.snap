// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightCard should match the App snapshot 1`] = `ShallowWrapper {}`;
